mlir/test/Dialect/Affine/loop-fusion-4.mlir

```diff
@@ -285,3 +285,32 @@ module {
     spirv.ReturnValue %3 : !spirv.array<8192 x f32>
   }
 }
+
+// -----
+
+// PRODUCER-CONSUMER-LABEL: func @same_memref_load_store
+func.func @same_memref_load_store(%producer : memref<32xf32>, %consumer: memref<16xf32>){
```

Comment:
Should we have a test with multiple stores inside consumer nest?

Reply:
Consumer nests stores aren't relevant to this change/testing. The dependence is between the producer store and consumer load. The consumer stores could be to different memrefs, and that's unrelated to this fix.

Reply:
Can you add an example with multiple producer stores then? You have replaced `scrStoreOpInst` with `storeOps`  in argument list of `createPrivateMemRef` function. A unit test reflecting this change will be helpful.

Reply:
Good point, but multiple producer stores (to the same memref) aren't supported for fusion. However, this utility still supports that handling. To exercise it, we'll have to add that support in fusion.

Reply:
Actually, multiple producer stores (to different memrefs) are supported. I'll find a test case to add here.

Reply:
Done, thanks.